src/digital_identity_service/share.session.fetch.result.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const Validation = require('../yoti_common/validation');
+const ShareQrCodeCreateResult = require('./share.qr.code.create.result');
+
+/**
+ * The share session fetch result
+ *
+ * @class ShareSessionFetchResult
+ */
+module.exports = class ShareSessionFetchResult {
+  /**
+   * @param {Object} response
+   */
+  constructor(response) {
+    Validation.isString(response.id, 'Session ID');
+    this.id = response.id;
+
+    Validation.isString(response.status, 'Status');
+    this.status = response.status;
+
+    Validation.isStringDate(response.created, 'Created');
+    this.created = new Date(response.created);
+
+    Validation.isStringDate(response.updated, 'Updated');
+    this.updated = new Date(response.updated);
+
+    if (response.expiry) {
+      Validation.isStringDate(response.expiry, 'Expiry');
+      this.expiry = new Date(response.expiry);
+    }
+
+    if (response.qrCode) {
+      Validation.isPlainObject(response.qrCode, 'QrCode');
+      this.qrCode = new ShareQrCodeCreateResult(response.qrCode);
```

Comment:
No!
We are only expecting an `id` to be in... and that would anyway not correspond to a creation....